cmd/theatre-envconsul/main.go

```diff
@@ -193,6 +219,25 @@ func mainError(ctx context.Context, command string) (err error) {
 			return errors.Wrap(err, "failed to decode envconsul environment variables")
 		}
 
+		// For every file reference, write the value resolved by envconsul to the path in
+		// filePaths
+		for key, value := range filePaths {
+			if value == pathTBD {
+				// generate file path prefixed by key
+				tempFilePath, err := ioutil.TempFile("", fmt.Sprintf("%s-*", key))
+				if err != nil {
+					return errors.Wrap(err, fmt.Sprintf("failed to write temporary file for key %s", key))
+				}
+				value = tempFilePath.Name()
+			}
+
+			// write file with value of envMap[key]
```

Comment:
Once we've done this, should we *unset* the environment variables that correspond to the `vault-file:KEY:/path` values, so that they're not exposed to the app that's exec'd?

Reply:
I think we could do that, or would it be an idea to set the value of the env var with the path where the secret now is written?

Reply:
I'd sort of assumed that we'd only be doing that for ones where we randomly generate the path. But actually, maybe this does make sense to do always!